homework-submission/todo-app-week1/src/App.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react'
+import { BrowserRouter, Route, Link } from 'react-router-dom'
+import ItemsComponent from './components/ItemsComponent'
+import './App.css'
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      todos: new TodosModel(),
+    };
+    this.update = this.update.bind(this);
+
+  }
+
+  completeItem = id => {
+    let items = {
+      ...this.state.items,
+      [id]: { ...this.state.items[id], completed: true }
+    }
+    this.setState({
+      items
+    })
+  }
+  deleteItem = id => {
+    let { [id]: deleted, ...items } = this.state.items
+    this.setState({
+      items
+    })
+  }
+  addItem = e => {
+    e.preventDefault()
+    let items = {
+      ...this.state.items,
+      [new Date().valueOf()]: {
```

Comment:
After review, from the beginning it seems a bit confusing for another side person who didn't work on this code.

Think about a larger application:
 - now you have like this:
```JSON
{
  "items": {
    "1541542946291": {
      "item": "hello world",
      "completed": false
    },
    "1541542951952": {
      "item": "do the dishes",
      "completed": false
    },
    "1541542958760": {
      "item": "do the laundry",
      "completed": true
    },
    "1541542965986": {
      "item": "make a shower ",
      "completed": true
    }
  }
}
```
 - in the near future in a real application you will have a very long JSON Object, rather than the lots of methods that are working all together to get the Id with a different methodology from each developer use.
In my opinion I think it's a bit cleaner to have an Array of Objects, so you will have the `new Date().valueOf()` as an `Id`, but still any way you will have it from the server.

In any case keep the huge application idea always in mind 🙂